### DISCUSSION TITLE
fatal: not a git repository

### DISCUSSION THREAD

**Discussion by SpectureCarrier** [bluecherrydvr/bluecherry-apps#419] [Q&A]
git submodule foreach --recursive 'git clean -dxf && git reset --hard'
fatal: not a git repository (or any of the parent directories): .git

**Comment by SpectureCarrier** [comment]
user@user-tpvm:~/Documents/bluecherry-apps-3.0.2$ su
Password: 
root@user-tpvm:/home/user/Documents/bluecherry-apps-3.0.2# scripts/build_pkg_native.sh
+ DST_DIR=releases//
+ [[ '' == centos* ]]
++ dirname scripts/build_pkg_native.sh
+ sudo scripts/install_prereqs_native.sh
+ for x in /etc/*-release
+ source /etc/lsb-release
++ DISTRIB_ID=Ubuntu
++ DISTRIB_RELEASE=18.04
++ DISTRIB_CODENAME=bionic
++ DISTRIB_DESCRIPTION='Ubuntu 18.04.5 LTS'
+ for x in /etc/*-release
+ source /etc/os-release
++ NAME=Ubuntu
++ VERSION='18.04.5 LTS (Bionic Beaver)'
++ ID=ubuntu
++ ID_LIKE=debian
++ PRETTY_NAME='Ubuntu 18.04.5 LTS'
++ VERSION_ID=18.04
++ HOME_URL=https://www.ubuntu.com/
++ SUPPORT_URL=https://help.ubuntu.com/
++ BUG_REPORT_URL=https://bugs.launchpad.net/ubuntu/
++ PRIVACY_POLICY_URL=https://www.ubuntu.com/legal/terms-and-policies/privacy-policy
++ VERSION_CODENAME=bionic
++ UBUNTU_CODENAME=bionic
+ [[ ubuntu == '' ]]
+ case $ID in
+ ADDITIONAL_PKGS='linux-image-generic linux-headers-generic'
+ case "`uname --hardware-platform`" in
++ uname --hardware-platform
+ [[ 18.04.5 LTS (Bionic Beaver) == \9\ \(\s\t\r\e\t\c\h\) ]]
+ [[ 18.04.5 LTS (Bionic Beaver) == \1\0\ \(\b\u\s\t\e\r\) ]]
+ ADDITIONAL_PKGS='linux-image-generic linux-headers-generic libmysqlclient-dev'
+ export DEBIAN_FRONTEND=noninteractive
+ DEBIAN_FRONTEND=noninteractive
+ apt-get update
Hit:1 https://mirrors.tuna.tsinghua.edu.cn/ubuntu bionic InRelease
Get:2 https://mirrors.tuna.tsinghua.edu.cn/ubuntu bionic-updates InRelease [88.7 kB]
Get:3 https://mirrors.tuna.tsinghua.edu.cn/ubuntu bionic-backports InRelease [74.6 kB]
Get:4 https://mirrors.tuna.tsinghua.edu.cn/ubuntu bionic-security InRelease [88.7 kB]
Get:5 https://mirrors.tuna.tsinghua.edu.cn/ubuntu bionic-proposed InRelease [242 kB]
Fetched 494 kB in 1s (408 kB/s)   
Reading package lists... Done
+ apt-get upgrade -y -V
Reading package lists... Done
Building dependency tree       
Reading state information... Done
Calculating upgrade... Done
The following packages were automatically installed and are no longer required:
   libllvm7 (1:7-3~ubuntu0.18.04.1)
   libllvm9 (1:9-2~ubuntu18.04.2)
Use 'sudo apt autoremove' to remove them.
The following packages have been kept back:
   linux-generic-hwe-18.04 (5.4.0.66.74~18.04.61 => 5.4.0.67.75~18.04.62)
   linux-headers-generic-hwe-18.04 (5.4.0.66.74~18.04.61 => 5.4.0.67.75~18.04.62)
   linux-image-generic-hwe-18.04 (5.4.0.66.74~18.04.61 => 5.4.0.67.75~18.04.62)
   netplan.io (0.40.1~18.04.4 => 0.99-0ubuntu3~18.04.4)
   python3-software-properties (0.96.24.32.7 => 0.96.24.32.14)
   software-properties-common (0.96.24.32.7 => 0.96.24.32.14)
   software-properties-gtk (0.96.24.32.7 => 0.96.24.32.14)
   ubuntu-drivers-common (1:0.5.2.2 => 1:0.8.6.3~0.18.04.1)
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
+ apt-get install -y -V autoconf automake libtool git build-essential gcc g++ debhelper ccache bison flex texinfo yasm cmake libbsd-dev libopencv-dev libudev-dev libva-dev rsyslog sudo linux-image-generic linux-headers-generic libmysqlclient-dev
Reading package lists... Done
Building dependency tree       
Reading state information... Done
autoconf is already the newest version (2.69-11).
automake is already the newest version (1:1.15.1-3ubuntu2).
bison is already the newest version (2:3.0.4.dfsg-1build1).
build-essential is already the newest version (12.4ubuntu1).
ccache is already the newest version (3.4.1-1).
flex is already the newest version (2.6.4-6).
libtool is already the newest version (2.4.6-2).
rsyslog is already the newest version (8.32.0-1ubuntu4).
libva-dev is already the newest version (2.1.0-3).
texinfo is already the newest version (6.5.0.dfsg.1-2).
yasm is already the newest version (1.3.0-2build1).
cmake is already the newest version (3.10.2-1ubuntu2.18.04.1).
debhelper is already the newest version (11.1.6ubuntu2).
g++ is already the newest version (4:7.4.0-1ubuntu2.3).
gcc is already the newest version (4:7.4.0-1ubuntu2.3).
git is already the newest version (1:2.17.1-1ubuntu0.8).
libbsd-dev is already the newest version (0.8.7-1ubuntu0.1).
libmysqlclient-dev is already the newest version (5.7.33-0ubuntu0.18.04.1).
sudo is already the newest version (1.8.21p2-3ubuntu1.4).
libopencv-dev is already the newest version (3.2.0+dfsg-4ubuntu0.1).
libudev-dev is already the newest version (237-3ubuntu10.45).
linux-headers-generic is already the newest version (4.15.0.137.124).
linux-image-generic is already the newest version (4.15.0.137.124).
The following packages were automatically installed and are no longer required:
   libllvm7 (1:7-3~ubuntu0.18.04.1)
   libllvm9 (1:9-2~ubuntu18.04.2)
Use 'sudo apt autoremove' to remove them.
0 upgraded, 0 newly installed, 0 to remove and 8 not upgraded.
+ git submodule foreach --recursive 'git clean -dxf && git reset --hard'
+ git submodule update --recursive --init
++ git describe --dirty --always --long
fatal: bad revision 'HEAD'
++ git describe --all
fatal: No names found, cannot describe anything.
+ echo '#define GIT_REVISION " "'
+ [[ '' == centos* ]]
+ debian/rules debian/control
sed -e 's/^#bionic//' debian/control.in | grep -v '^#' > debian/control
+ debian/rules clean binary
dh --parallel clean
dh: Compatibility levels before 9 are deprecated (level 7 in use)
   dh_auto_clean
dh_auto_clean: Compatibility levels before 9 are deprecated (level 7 in use)
	make -j1 clean
make[1]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2'
make[2]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/misc'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
make -C libconfig clean
make[3]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/libconfig'
make[3]: *** No rule to make target 'clean'.  Stop.
make[3]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/libconfig'
BCMK:15: recipe for target 'libconfig-clean' failed
make[2]: [libconfig-clean] Error 2 (ignored)
rm libconfig/Makefile
rm: cannot remove 'libconfig/Makefile': No such file or directory
BCMK:15: recipe for target 'libconfig-clean' failed
make[2]: [libconfig-clean] Error 1 (ignored)
make -C libav distclean
make[3]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/libav'
make[3]: *** No rule to make target 'distclean'.  Stop.
make[3]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/libav'
BCMK:122: recipe for target 'libav-clean' failed
make[2]: [libav-clean] Error 2 (ignored)
make -C pugixml clean
make[3]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/pugixml'
make[3]: *** No rule to make target 'clean'.  Stop.
make[3]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/misc/pugixml'
BCMK:150: recipe for target 'pugixml-clean' failed
make[2]: [pugixml-clean] Error 2 (ignored)
rm pugixml/Makefile
rm: cannot remove 'pugixml/Makefile': No such file or directory
BCMK:150: recipe for target 'pugixml-clean' failed
make[2]: [pugixml-clean] Error 1 (ignored)
rm -rf /home/user/Documents/bluecherry-apps-3.0.2/stage
make[2]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/misc'
make[2]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/lib'
rm -f bc-core.o bc-utils.o bc-db-core.o bc-db-mysql.o bc-key.o bc-media.o lavf_device.o bc-ptz.o input_device.o v4l2_device_solo6x10.o stream_elements.o logging.o bc-syslog.o sliding_seq_window.o v4l2_device_tw5864.o v4l2_device_solo6010-dkms.o bc-core.d bc-utils.d bc-db-core.d bc-db-mysql.d bc-key.d bc-media.d lavf_device.d bc-ptz.d input_device.d v4l2_device_solo6x10.d stream_elements.d logging.d bc-syslog.d sliding_seq_window.d v4l2_device_tw5864.d v4l2_device_solo6010-dkms.d libbluecherry.so.0 libbluecherry.so
make[2]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/lib'
make[2]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/server'
rm -f bc-server.o bc-thread.o media_writer.o g723-dec.o bc-detect.o streaming.o rtsp.o signals.o motion_processor.o trigger_processor.o decoder.o encoder.o reencoder.o vaapi.o scaler.o recorder.o motion_handler.o ffmpeg-init.o bt.o status_server.o trigger_server.o substream-thread.o bc-server.d bc-thread.d media_writer.d g723-dec.d bc-detect.d streaming.d rtsp.d signals.d motion_processor.d trigger_processor.d decoder.d encoder.d reencoder.d vaapi.d scaler.d recorder.d motion_handler.d ffmpeg-init.d bt.d status_server.d trigger_server.d substream-thread.d bc-server
make[2]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/server'
make[2]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/utils'
rm -f pelcocmd ptzcmd bc-license licensecmd *.o
make[2]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/utils'
make[1]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2'
   dh_clean
dh_clean: Compatibility levels before 9 are deprecated (level 7 in use)
dh --parallel binary
dh: Compatibility levels before 9 are deprecated (level 7 in use)
   dh_update_autotools_config
   dh_auto_configure
dh_auto_configure: Compatibility levels before 9 are deprecated (level 7 in use)
   dh_auto_build
dh_auto_build: Compatibility levels before 9 are deprecated (level 7 in use)
	make -j1
make[1]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2'
make[2]: Entering directory '/home/user/Documents/bluecherry-apps-3.0.2/misc'
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
fatal: No names found, cannot describe anything.
cd libconfig/ && autoreconf -if
autoreconf: 'configure.ac' or 'configure.in' is required
BCMK:19: recipe for target 'libconfig/Makefile' failed
make[2]: *** [libconfig/Makefile] Error 1
make[2]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2/misc'
Makefile:20: recipe for target 'misc/all' failed
make[1]: *** [misc/all] Error 2
make[1]: Leaving directory '/home/user/Documents/bluecherry-apps-3.0.2'
dh_auto_build: make -j1 returned exit code 2
debian/rules:12: recipe for target 'binary' failed
make: *** [binary] Error 2
root@user-tpvm:/home/user/Documents/bluecherry-apps-3.0.2#

**Comment by sicada** [comment]
The **build_pkg_native.sh** script does not clone the git repository; it assumes you've already done that. All of the files in your **/home/user/Documents/bluecherry-apps-3.0.2** folder should have come from github via a clone command, such as:

`git clone --progress --depth 1 http://github.com/bluecherrydvr/bluecherry-apps.git`

If you don't have a folder named **.git** in **/home/user/Documents/bluecherry-apps-3.0.2/** then you didn't properly clone the repository.
